From 6fde4e1a9c486960860238a0afeebb5bc68b9493 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Sun, 10 Mar 2024 21:25:08 +0100
Subject: [PATCH 10/11] scripts/buildtools.mk: style suggestions

* $(filter nostrip,$(DEB_BUILD_OPTIONS)) is slightly better than
  $(findstring) because the latter matches 'arduinostrip'.

* An explicit test of $(origin) instead of ?= makes the intent of each
  stanza more readable (first ensure that TOOL is set, then
  TOOL_FOR_BUILT).

* Special case nostrip outside generic dpkg_buildtool_setvar
  procedure.
---
 scripts/mk/buildtools.mk | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/scripts/mk/buildtools.mk b/scripts/mk/buildtools.mk
index d70f546f5..aefef9b47 100644
--- a/scripts/mk/buildtools.mk
+++ b/scripts/mk/buildtools.mk
@@ -39,22 +39,17 @@ include $(dpkg_datadir)/architecture.mk
 # not defined or contain the make built-in defaults. On native builds if
 # TOOL is defined and TOOL_FOR_BUILD is not, we fallback to use TOOL.
 define dpkg_buildtool_setvar
-  ifeq (,$(findstring $(3),$(DEB_BUILD_OPTIONS)))
-    ifeq ($(origin $(1)),default)
-      $(1) = $(DEB_HOST_GNU_TYPE)-$(2)
-    else
-      $(1) ?= $(DEB_HOST_GNU_TYPE)-$(2)
-    endif
+  ifneq (,$(filter default undefined,$(origin $(1))))
+    $(1) = $(DEB_HOST_GNU_TYPE)-$(2)
+  endif
 
+  ifeq (undefined,$(origin $(1)_FOR_BUILD))
     # On native build fallback to use TOOL if that's defined.
     ifeq ($(DEB_BUILD_GNU_TYPE),$(DEB_HOST_GNU_TYPE))
-      $(1)_FOR_BUILD ?= $$($(1))
+      $(1)_FOR_BUILD = $$($(1))
     else
-      $(1)_FOR_BUILD ?= $(DEB_BUILD_GNU_TYPE)-$(2)
+      $(1)_FOR_BUILD = $(DEB_BUILD_GNU_TYPE)-$(2)
     endif
-  else
-    $(1) = :
-    $(1)_FOR_BUILD = :
   endif
 
   ifdef DPKG_EXPORT_BUILDTOOLS
@@ -72,7 +67,12 @@ $(eval $(call dpkg_buildtool_setvar,OBJCXX,g++))
 $(eval $(call dpkg_buildtool_setvar,F77,gfortran))
 $(eval $(call dpkg_buildtool_setvar,FC,gfortran))
 $(eval $(call dpkg_buildtool_setvar,LD,ld))
-$(eval $(call dpkg_buildtool_setvar,STRIP,strip,nostrip))
+ifneq (,$(filter nostrip,$(DEB_BUILD_OPTIONS)))
+  STRIP = :
+  STRIP_FOR_BUILD = :
+  # But call dpkg_buildtool_setvar for the export.
+endif
+$(eval $(call dpkg_buildtool_setvar,STRIP,strip))
 $(eval $(call dpkg_buildtool_setvar,OBJCOPY,objcopy))
 $(eval $(call dpkg_buildtool_setvar,OBJDUMP,objdump))
 $(eval $(call dpkg_buildtool_setvar,NM,nm))
-- 
2.39.2

